[EuiEmptyPrompt] Add more guidelines

miukimiu opened the pull request:
### Summary

This PR adds more usages guidelines for the **EuiEmptyPrompt**. You can find the page for reviewing here: https://eui.elastic.co/pr_5409/#/display/empty-prompt/guidelines.

### Guidelines

There were some discussions that can be found on this [google doc](https://docs.google.com/document/d/1k0i1s5bli-Gg2og_7VEBUqcP8sLthjReaKYGyLnQ5u0/edit#) that lead to some decisions. 

### Panel color picker

The panel color picker is a component where based on the selection of a use case and a page template a panel color is recommended. 

I used to have an example for "completed tasks" but I removed it. I couldn't think of a good example. But if anyone has a good example, I'm happy to add it. @gchaps do you have a good example of this use case?

Initially, alternatives were also shown but I simplified to only show one recommendation. 

The "Default with multiple panels" page template option gets disabled for:

* No permission
* Page not found 
* Loading

<img width="1168" alt="Screenshot 2021-12-17 at 15 22 21" src="https://user-images.githubusercontent.com/2750668/146567284-fe97c3da-d54b-4b88-a032-4146ccde63f2.png">

### Guide Section component

The guide section component was improved so it can receive JS strings. We could only pass RAW js files that were imported like `require('!!raw-loader!..)`. This way I can use the guide section to preview the result of the panel color picker.

<img width="784" alt="Screenshot 2021-12-16 at 20 21 32" src="https://user-images.githubusercontent.com/2750668/146567234-e5ec1b67-b7d5-4827-8b8a-2d2322550fe9.png">


I'm using a component [react-element-to-jsx-string](https://github.com/algolia/react-element-to-jsx-string) to transform generated react components to a JSX string and this is passed to the demo js tab.

But... I found a weird issue. It doesn't work well when deployed. As we can see the deployed version gets some weird tags. 😬 

<img width="1254" alt="plugin@2x" src="https://user-images.githubusercontent.com/2750668/146563386-b8b4c767-62e6-41a1-88f4-60bfd68fefdb.png">

It was recommended to use **react-view** but I found it difficult to achieve what I was achieving with **react-element-to-jsx-string**. @thompsongl can you help me with this task?


### Checklist

- [x] Check against **all themes** for compatibility in both light and dark modes
- ~[ ] Checked in **mobile**~
- [x] Checked in **Chrome**, **Safari**, **Edge**, and **Firefox**
- ~[ ] Props have proper **autodocs** and **[playground toggles](https://github.com/elastic/eui/blob/main/wiki/documentation-guidelines.md#adding-playground-toggles)**~
- [x] Added **[documentation](https://github.com/elastic/eui/blob/main/wiki/documentation-guidelines.md)**
- ~[ ] Checked **[Code Sandbox](https://codesandbox.io/)** works for any docs examples~
- ~[ ] Added or updated **[jest](https://github.com/elastic/eui/blob/main/wiki/testing.md) and [cypress](https://github.com/elastic/eui/blob/main/wiki/cypress-testing.md) tests**~
- ~[ ] Checked for **breaking changes** and labeled appropriately~
- [x] Checked for **accessibility** including keyboard-only and screenreader modes
- ~[ ] A **[changelog](https://github.com/elastic/eui/blob/main/wiki/documentation-guidelines.md#changelog)** entry exists and is marked appropriately~

kibanamachine commented:
Preview documentation changes for this PR: https://eui.elastic.co/pr_5409/

kibanamachine commented:
Preview documentation changes for this PR: https://eui.elastic.co/pr_5409/

kibanamachine commented:
Preview documentation changes for this PR: https://eui.elastic.co/pr_5409/

kibanamachine commented:
Preview documentation changes for this PR: https://eui.elastic.co/pr_5409/

kibanamachine commented:
Preview documentation changes for this PR: https://eui.elastic.co/pr_5409/

kibanamachine commented:
Preview documentation changes for this PR: https://eui.elastic.co/pr_5409/

kibanamachine commented:
Preview documentation changes for this PR: https://eui.elastic.co/pr_5409/

kibanamachine commented:
Preview documentation changes for this PR: https://eui.elastic.co/pr_5409/

kibanamachine commented:
Preview documentation changes for this PR: https://eui.elastic.co/pr_5409/

kibanamachine commented:
Preview documentation changes for this PR: https://eui.elastic.co/pr_5409/

kibanamachine commented:
Preview documentation changes for this PR: https://eui.elastic.co/pr_5409/

miukimiu commented:
jenkins test this

kibanamachine commented:
Preview documentation changes for this PR: https://eui.elastic.co/pr_5409/

kibanamachine commented:
Preview documentation changes for this PR: https://eui.elastic.co/pr_5409/

kibanamachine commented:
Preview documentation changes for this PR: https://eui.elastic.co/pr_5409/

cristina-eleonora commented:
This looks great, @miukimiu ! Love the interactive section for the panel colors ❤️ 

It would be great if we also had an example for good empty prompts where users cannot take any actions, for example: you're a new customer and you don't have any invoices. 

Could we also add an example for how a 403 page might look like? In cloud, we recently designed it to look like this:

<img width="746" alt="Screenshot 2021-12-17 at 17 42 50" src="https://user-images.githubusercontent.com/44808440/146578597-019607c0-137d-47d7-b4b4-8e9be302e0e0.png">

For the 404 page, should we also mention that if a page is unavailable, it may have never existed in the first place (e.g.: there may be a typo in the text). E.g.:

`The page you are looking for might have been removed, renamed, or it did not exists in the first place.`. Cc @gchaps

miukimiu commented:
Thanks, @cristina-eleonora. I added a new example of how a 403 page might look like. For the 404 example, I'm happy to improve the text based on your recommendation. But let's see what @gchaps thinks.

kibanamachine commented:
Preview documentation changes for this PR: https://eui.elastic.co/pr_5409/

cchaos reviewed (commented):
These are going to be super helpful! I'm going to push up a commit where I've converted your custom buttons to use the EuiKeyPadMenu instead. It'll have all the right accessibility requirements for that kind of radio group. 

----

I found myself having to cross-reference the "types" table and the panel recommendations. What do you think about just combining them into this layout you have by putting the contents of the table in the right split panel when the selection is made?
![Image 2022-01-05 at 4 30 29 PM](https://user-images.githubusercontent.com/549577/148292567-88025c8e-5142-4596-a8cd-c370c83f2465.jpg)

---

A couple of notes/questions on the specific use cases:

**1. No Data**
I'm thinking about how we really want to present this example. Currently it doesn't align with the implementation we established in Kibana for 7.16 which uses an EuiCard.

**2. No permission**
We should align this with the current doc that Monina started. https://docs.google.com/document/d/1Tio1LAxHMOax7J45_nsVldT1exoXavKgHKA_0QpTvyQ/edit

I'm also not sure why there is a difference between "No permission" and "Access denied".

**3. No results**

The example that is showing is less like no **results** than no **created things**. The term results to me is that it was something that the user **searched** for. How about "No items yet" or something?

**4. Page not found**

I don't think the searchExclamation icon works in this situation. I'd opt for leaving it off or putting in an illustration.

**5. Loading**

This one can also be possible in the multi-panel situation but likely we'd want a more subdued version like using the spinner instead of logo loader.

In fact for all of the multi-panel scenarios, I'd decrease the `titleSize`.

----

It looks like that issue in production with the JS string converter is still there. Were you able to look into `react-view` like how the Playground does it?

gchaps commented on src-docs/src/views/empty_prompt/guidelines.tsx:
```diff
@@ -188,13 +213,177 @@ export default () => (
+        <>
+          <p>
+            Icons and illustrations must first and foremost communicate meaning.
+            They are also an opportunity to delight users and show our Elastic
```
If these are guidelines for more than just Elastic, this should say "show your brand" instead of "show our Elastic brand."

gchaps commented on src-docs/src/views/empty_prompt/guidelines.tsx:
```diff
@@ -188,13 +213,177 @@ export default () => (
+      </GuideRuleExample>
+      <GuideRuleExample
+        type="dont"
+        text="Avoid using icons and illustrations that don't mean anything and are not related with the content."
```
```suggestion
        text="Avoid using icons and illustrations that don't mean anything and are not related to the content."
```

gchaps commented on src-docs/src/views/empty_prompt/guidelines.tsx:
```diff
@@ -188,13 +213,177 @@ export default () => (
+      description={
+        <>
+          <p>
+            Use learn more links in your empty prompt to link to documentation
```
Suggest using quotation marks to make the text stand out:

Use "Learn more" links...

gchaps commented on src-docs/src/views/empty_prompt/guidelines.tsx:
```diff
@@ -188,13 +213,177 @@ export default () => (
+            where users can get more detailed help.
+          </p>
+          <p>
+            Use the link after the description when the empty prompt
```
```suggestion
            Include the link after the description when the empty prompt
```

gchaps commented on src-docs/src/views/empty_prompt/guidelines.tsx:
```diff
@@ -188,13 +213,177 @@ export default () => (
+          </p>
+          <p>
+            Use the link after the description when the empty prompt
+            doesn&apos;t contain any call to action. Otherwise, if there is a
```
```suggestion
            doesn&apos;t contain a call to action. If there is a
```

gchaps commented on src-docs/src/views/empty_prompt/guidelines.tsx:
```diff
@@ -188,13 +213,177 @@ export default () => (
+          <p>
+            Use the link after the description when the empty prompt
+            doesn&apos;t contain any call to action. Otherwise, if there is a
+            call to action, use the learn more link in the footer.
```
```suggestion
            call to action, include the link in the footer.
```

gchaps commented on src-docs/src/views/empty_prompt/guidelines.tsx:
```diff
@@ -188,13 +213,177 @@ export default () => (
+      description={
+        <>
+          <p>
+            When having multiple empty states throughout a page, avoid having
```
```suggestion
            When a page has multiple empty states, avoid using
```

gchaps commented on src-docs/src/views/empty_prompt/guidelines.tsx:
```diff
@@ -188,13 +213,177 @@ export default () => (
+        <>
+          <p>
+            When having multiple empty states throughout a page, avoid having
+            multiple primary actions and multiple icons/illustrations.
```
```suggestion
            multiple primary actions and multiple icons or illustrations.
```

gchaps commented on src-docs/src/views/empty_prompt/guidelines.tsx:
```diff
@@ -188,13 +213,177 @@ export default () => (
+            multiple primary actions and multiple icons/illustrations.
+          </p>
+          <p>
+            Use secondary actions and no icons/illustrations. This way, the
```
```suggestion
            Use secondary actions and no icons or illustrations. This way, the
```

gchaps commented on src-docs/src/views/empty_prompt/guidelines.tsx:
```diff
@@ -188,13 +213,177 @@ export default () => (
+    >
+      <GuideRuleExample
+        type="do"
+        text="Use secondary actions and no icons/illustrations when displaying multiple empty states."
```
```suggestion
        text="Use secondary actions and no icons or illustrations when displaying multiple empty states."
```

kibanamachine commented:
Preview documentation changes for this PR: https://eui.elastic.co/pr_5409/

gchaps commented on src-docs/src/views/empty_prompt/guidelines.tsx:
```diff
@@ -27,58 +32,62 @@ export default () => (
     {/* End of Content section */}
-    {/* Usage section */}
+    {/* Types of empty states and goals section */}
     <GuideRuleTitle>Types of empty states and goals</GuideRuleTitle>
```
It would be really helpful to include an example of each type of empty stated listed here.

cchaos replied:
I think this comment aligns with my suggestion about just moving the table contents to the panel colors section to combine them all into one section.

kibanamachine commented:
Preview documentation changes for this PR: https://eui.elastic.co/pr_5409/

kyrspl commented:
Thanks @miukimiu - We're applying this to the cloud UI in [94718](https://github.com/elastic/cloud/pull/94718)

miukimiu commented:
I'm still working on this PR but just to give a few updates. 

- I talked with @chandler about the usage of [react-element-to-jsx-string](https://github.com/algolia/react-element-to-jsx-string). I did an attempt to use **react-view** but it was not straightforward. We agreed that an engineer is going to help me with this task.
- The table and panel colors section are now combined. The texts are reflecting @gchaps suggestions. But I still need to go through the "couple of notes/questions on the specific use cases" that @cchaos added.
-  403 and 404 texts are following this suggestion (https://github.com/elastic/cloud/issues/93371#issuecomment-1006075870 )
- Tried to use illustrations for the 403 and 404 use cases but I might **revert this change**. The illustrations are being used in **cloud** as PNGs. I was able to get the illustrations as SVGs but when I optimize them they end up with 2.9MB and 4.7MB. So I guess that's the reason in Cloud they're using the images as PNGs.
- The build is failing (https://kibana-ci.elastic.co/job/elastic+eui+pull-request-test/5802/console). Is it because of the PNGs that I'm importing? 🤔

kibanamachine commented:
Preview documentation changes for this PR: https://eui.elastic.co/pr_5409/

kibanamachine commented:
Preview documentation changes for this PR: https://eui.elastic.co/pr_5409/

chandlerprall commented:
> Tried to use illustrations for the 403 and 404 use cases but I might revert this change. The illustrations are being used in cloud as PNGs. I was able to get the illustrations as SVGs but when I optimize them they end up with 2.9MB and 4.7MB. So I guess that's the reason in Cloud they're using the images as PNGs.

I've tracked down the configuration that causes this (it comes from how our typescript->proptypes babel plugin interacts with the `@babel/typescript` preset), but haven't figured out how to avoid the situation. Digging further.

kibanamachine commented:
Preview documentation changes for this PR: https://eui.elastic.co/pr_5409/

chandlerprall commented:
@miukimiu resolved the issue with importing PNGs, and I made TS happy with renaming __types_of_use_cases_ back to typescript. I'll look at the JSX generation tomorrow

kibanamachine commented:
Preview documentation changes for this PR: https://eui.elastic.co/pr_5409/

kibanamachine commented:
Preview documentation changes for this PR: https://eui.elastic.co/pr_5409/

kibanamachine commented:
Preview documentation changes for this PR: https://eui.elastic.co/pr_5409/

kibanamachine commented:
Preview documentation changes for this PR: https://eui.elastic.co/pr_5409/

chandlerprall commented:
Started down the road to use `react-view` for formatting. First snag was the inability to nest component configs, instead you'd have to configure the innermost component, format the code, and inject that string into its parent component usage, repeating all the way up.

Okay, let's give that a shot. Second snag: there's a lot of components in play here and the various attributes come and go, and all of that would need to be accounted for. Here's where I got to before pulling up for air:

**Building just the empty prompt**

Before
```
const icon = currentUseCaseExample.iconType
    ? { iconType: currentUseCaseExample.iconType }
    : {
        icon: currentUseCaseExample.icon,
      };

  const layout = currentUseCaseExample.layout
    ? { layout: currentUseCaseExample.layout }
    : { layout: 'vertical' };

  const euiEmptyPromptPreview = (
    <EuiEmptyPrompt
      {...icon}
      {...layout}
      title={currentUseCaseExample.title}
      body={currentUseCaseExample?.body}
      actions={currentUseCaseExample?.actions}
      {...(panelProps as any)}
    />
  );
```

After (which does not implement spreading `panelProps`, so this is not complete)
```
const euiEmptyPromptPreviewIconProp: TUseViewParams['props'] = currentUseCaseExample.iconType
    ? {
        iconType: {
          type: PropTypes.String,
          value: currentUseCaseExample.iconType,
          description: '',
        },
      }
    : {
        icon: {
          type: PropTypes.String,
          value: currentUseCaseExample.icon,
          description: '',
        },
      };
  const euiEmptyPromptPreview = useView({
    componentName: 'EuiEmptyPrompt',
    props: {
      ...euiEmptyPromptPreviewIconProp,
      layout: {
        type: PropTypes.String,
        value: currentUseCaseExample.layout ?? 'vertical',
        description: '',
      },
      title: {
        type: PropTypes.ReactNode,
        value: currentUseCaseExample.title,
        description: '',
      },
      body: {
        type: PropTypes.ReactNode,
        value: currentUseCaseExample?.body ?? undefined,
        description: '',
      },
      actions: {
        type: PropTypes.Array,
        value: currentUseCaseExample?.actions,
        description: '',
      },
    },
    scope: viewScope,
  });
```

Another config would need to be created for each of the demo types: sidebar, no-sidebar, multiple panels. The multiple panel example type nests a [number of layout components](https://github.com/elastic/eui/blob/e87edebf05169ff10f61a9b6fc9ec4da0b0d7d7f/src-docs/src/views/empty_prompt/_types_of_empty_states.tsx#L200-L232), and each level would need a config->code->getSnippet(cleanup) pass.

I don't think this is a worthwhile conversion, unless we wanted to explore using all of `react-view`'s playground functionality + custom knobs/controls to build up these wizard-like component builders.

miukimiu commented:
Thanks, @chandlerprall for trying to fix this issue.

> I don't think this is a worthwhile conversion

I agree. It needs to be easy to transform a component in a jsx string. So you think the solution is to use **react-element-to-jsx-string** with a better webpack config?

kibanamachine commented:
Preview documentation changes for this PR: https://eui.elastic.co/pr_5409/

chandlerprall commented:
Webpack change pushed. I opted to keep _all_ **Eui*** function component names from being mangled, instead of a specific list. For the _**docs**_ production build this means an increase bundle size of 9348 bytes (non-gzipped difference).

kibanamachine commented:
Preview documentation changes for this PR: https://eui.elastic.co/pr_5409/

kibanamachine commented:
Preview documentation changes for this PR: https://eui.elastic.co/pr_5409/

kibanamachine commented:
Preview documentation changes for this PR: https://eui.elastic.co/pr_5409/

miukimiu commented:
Thanks, @chandlerprall for fixing the DemoJS tab and the issue with importing PNGs. 🎉 

@cchaos, I went through all your comments and the PR is ready for review. 🙌🏽

cchaos reviewed (commented):
Overall LGTM. I just found a couple bugs and have minor copy edits to finish it off.

1. I'm getting `icon={<No Display Name />}` in the Demo JS (specifically when using an image as the icon)

kibanamachine commented:
Preview documentation changes for this PR: https://eui.elastic.co/pr_5409/

kibanamachine commented:
Preview documentation changes for this PR: https://eui.elastic.co/pr_5409/

miukimiu commented on src-docs/src/views/empty_prompt/_types_of_use_cases.tsx:
```diff
@@ -0,0 +1,307 @@
+  return <EuiImage size="fullWidth" src={pageNotFoundImg} alt="" />;
+};
+
+// Forces webpack to show the display name.
+// Required to show the name of the component in the DemoJS tab instead of `<No Display Name />`.
+PageNotFoundImg.displayName = 'PageNotFoundImg';
```
> I'm getting icon={<No Display Name />} in the Demo JS (specifically when using an image as the icon)

I fixed this issue by adding the `displayName`. @chandlerprall is this the right way to solve this issue?

chandlerprall replied:
Yep, that's perfect!

cchaos replied:
Why isn't is showing up though as `<EuiImage url="..." />` like the `pageSideBar` prop value is?
![Screen Shot 2022-02-09 at 14 45 38 PM](https://user-images.githubusercontent.com/549577/153278420-cbac91cb-b78f-46ec-9900-ccd3d7e6e177.png)

miukimiu replied:
Because it is a component. 

```js
const PageNotFoundImg = () => {
  const themeContext = useContext(ThemeContext);
  const isDarkTheme = themeContext.theme.includes('dark');

  const pageNotFoundImg = isDarkTheme ? pageNotFoundDark : pageNotFoundLight;

  return <EuiImage size="fullWidth" src={pageNotFoundImg} alt="" />;
};
```

I had to create a component to use the `useContext` hook. I couldn't use it inside the object.

cchaos replied:
I'm just thinking if people copy/paste, they might think it's a real EUI component.

miukimiu replied:
Maybe there is a better way of handling this. Let me try a better approach.

miukimiu replied:
@cchaos, the issue was fixed in https://github.com/elastic/eui/pull/5409/commits/04449ecaa55cfd8ccb1e2a946a8fbeebf4ac89db.

miukimiu commented:
Thanks, @cchaos for helping me out with the copy. 🎉 

I fixed the error below. I just need @chandlerprall to confirm if the way I solved it is ok. 

>I'm getting icon={<No Display Name />} in the Demo JS (specifically when using an image as the icon)

I also replaced the thumbnail for the "learn more" link.

kibanamachine commented:
Preview documentation changes for this PR: https://eui.elastic.co/pr_5409/

chandlerprall commented on scripts/babel/proptypes-from-ts-props/index.test.ts:
```diff
@@ -2514,6 +2514,35 @@ let something: any;
+          }
+        );
+
+        expect(result.code).toBe("export {};"); // code is empty as some plugin
```
That change is great; this was impacted by the recent change (#5554) to use the `export type` syntax so the babel typescript plugin would remove type exports during the build, instead doing that with a lot of code in our proptypes plugin.

I'll push a commit to remove the comment from that `expect` line; it was an incomplete thought (oops) why the expected string was empty as opposed to the `export {}` we'd otherwise expect (as this needs to now).

kibanamachine commented:
Preview documentation changes for this PR: https://eui.elastic.co/pr_5409/

cchaos commented on src-docs/src/views/empty_prompt/guidelines.tsx:
```diff
@@ -27,174 +28,210 @@ export default () => (
+          space is empty and guide the users through the required actions.
+        </li>
+        <li>
+          <strong>Action(s):</strong> Your call to actions. Should answer the
+          question What will solve the issue? Lead the users to take action or
+          guide them about the next steps.
```
Looks like this one was missed ^^

gchaps commented on src-docs/src/views/empty_prompt/guidelines.tsx:
```diff
@@ -27,174 +28,210 @@ export default () => (
+    <GuideRuleTitle>Design</GuideRuleTitle>
+
+    <GuideRule
+      heading="Vertical vs. Horizontal"
```
```suggestion
      heading="Vertical vs. horizontal"
```

gchaps commented on src-docs/src/views/empty_prompt/guidelines.tsx:
```diff
@@ -27,174 +28,210 @@ export default () => (
+        <>
+          <p>
+            The vertical layout is perfect when the content is small—a title and
+            two paragraphs at most. You can use this layout with just an icon,
```
```suggestion
            two paragraphs at most. You can use this layout with an icon,
```

gchaps replied:
I think this should be "two sentences at most".  Two paragraphs is a lot of text.

gchaps commented on src-docs/src/views/empty_prompt/guidelines.tsx:
```diff
@@ -27,174 +28,210 @@ export default () => (
+          </p>
+          <p>
+            Use the horizontal layout when you have a long description, multiple
+            calls to action, and possibly a footer. For this type of layout, an
```
```suggestion
            calls to action, and a footer. For this type of layout, an
```

gchaps commented on src-docs/src/views/empty_prompt/guidelines.tsx:
```diff
@@ -27,174 +28,210 @@ export default () => (
+      }
+    >
+      <GuideRuleExample
+        text="Use the vertical layout when the content is small."
```
If you go with the text above, maybe this should be "Use the vertical layout when the text is 1 to 2 sentences."

gchaps commented on src-docs/src/views/empty_prompt/guidelines.tsx:
```diff
@@ -27,174 +28,210 @@ export default () => (
+        <EuiImage alt="Inline link" url={inlineLink} height="252" />
+      </GuideRuleExample>
+      <GuideRuleExample
+        text='Add the "learn more" link in the footer when the empty prompt contains a call to action.'
```
```suggestion
        text='Add the "Learn more" link in the footer when the empty prompt contains a call to action.'
```

gchaps commented on src-docs/src/views/empty_prompt/guidelines.tsx:
```diff
@@ -27,174 +28,210 @@ export default () => (
+      }
+    >
+      <GuideRuleExample
+        text='Add the "learn more" link after the description when the empty prompt doesn&apos;t contain a call to action.'
```
```suggestion
        text='Add the "Learn more" link after the description when the empty prompt doesn&apos;t contain a call to action.'
```

gchaps commented on src-docs/src/views/empty_prompt/guidelines.tsx:
```diff
@@ -27,174 +28,210 @@ export default () => (
+      description={
+        <>
+          <p>
+            Use &quot;learn more&quot; links in your empty prompt to link to
```
```suggestion
            Use &quot;Learn more&quot; links in your empty prompt to link to
```

gchaps replied:
I think we should mention that "Learn more" links are optional.

gchaps commented on src-docs/src/views/empty_prompt/guidelines.tsx:
```diff
@@ -27,174 +28,210 @@ export default () => (
+
+      <GuideRuleExample
+        type="dont"
+        text="Avoid mixing different types of empty states. Try to be consistent."
```
```suggestion
        text="Avoid mixing different types of empty states. Be consistent."
```

gchaps commented on src-docs/src/views/empty_prompt/_types_of_empty_states.tsx:
```diff
@@ -0,0 +1,399 @@
+            )}
+
+            <EuiDescriptionListTitle>
+              Recommend panel color
```
```suggestion
              Recommended panel color
```

gchaps commented on src-docs/src/views/empty_prompt/_types_of_use_cases.tsx:
```diff
@@ -0,0 +1,284 @@
+      goal: (
+        <p>
+          Help users understand how they can start using the product. For no
+          data use cases consider using another component.
```
```suggestion
          data use cases, consider using another component.
```

gchaps replied:
Can you suggest what the other component should be?

gchaps commented on src-docs/src/views/empty_prompt/_types_of_use_cases.tsx:
```diff
@@ -0,0 +1,284 @@
+      title: <h2>Start adding cases</h2>,
+      body: (
+        <p>
+          There are no cases to display. Add a new case or change your filter
```
I don't think "There are no cases to display" is necessary.

gchaps commented on src-docs/src/views/empty_prompt/_types_of_use_cases.tsx:
```diff
@@ -0,0 +1,284 @@
+        </EuiFlexItem>
+        <EuiFlexItem grow={false}>
+          <EuiTitle size="xxs">
+            <h3>Pretty sure you have data?</h3>
```
```suggestion
            <h3>Sure you have data?</h3>
```

gchaps commented on src-docs/src/views/empty_prompt/_types_of_use_cases.tsx:
```diff
@@ -0,0 +1,284 @@
+          and what actions to take to get access.
+        </p>
+      ),
+      action: <p>Request permission</p>,
```
```suggestion
      action: <p>Request permission.</p>,
```

gchaps commented on src-docs/src/views/empty_prompt/_types_of_use_cases.tsx:
```diff
@@ -0,0 +1,284 @@
+    },
+    example: {
+      iconType: 'alert',
+      title: <h2>Error loading Dashboards</h2>,
```
```suggestion
      title: <h2>Unable to load your dashboards</h2>,
```

gchaps commented on src-docs/src/views/empty_prompt/_types_of_use_cases.tsx:
```diff
@@ -0,0 +1,284 @@
+      title: <h2>Error loading Dashboards</h2>,
+      body: (
+        <p>
+          There was an error loading the Dashboard application. Contact your
```
```suggestion
          There was a problem loading the Dashboard application. Contact your
```

gchaps commented on src-docs/src/views/empty_prompt/_types_of_use_cases.tsx:
```diff
@@ -0,0 +1,284 @@
+        </p>
+      ),
+      goal: <p>Help users understand there is a client or server error.</p>,
+      action: <p>Go home or go back</p>,
```
```suggestion
      action: <p>Go home or go back.</p>,
```

gchaps commented on src-docs/src/views/empty_prompt/_types_of_use_cases.tsx:
```diff
@@ -0,0 +1,284 @@
+      body: (
+        <p>
+          Sorry, we can&apos;t seem to find the page you&apos;re looking for. It
+          might have been removed or renamed, or maybe it never existed at all.
```
```suggestion
          might have been removed or renamed, or maybe it never existed.
```

gchaps commented on src-docs/src/views/empty_prompt/_types_of_use_cases.tsx:
```diff
@@ -0,0 +1,284 @@
+      title: <h2>Page not found</h2>,
+      body: (
+        <p>
+          Sorry, we can&apos;t seem to find the page you&apos;re looking for. It
```
```suggestion
          Sorry, we can&apos;t find the page you&apos;re looking for. It
```

gchaps commented on src-docs/src/views/empty_prompt/_types_of_use_cases.tsx:
```diff
@@ -0,0 +1,284 @@
+          the license.
+        </p>
+      ),
+      action: <p>Start a trial or upgrade the license</p>,
```
```suggestion
      action: <p>Start a trial or upgrade the license.</p>,
```

gchaps commented on src-docs/src/views/empty_prompt/_types_of_use_cases.tsx:
```diff
@@ -0,0 +1,284 @@
+    },
+    example: {
+      iconLoading: <EuiLoadingLogo logo="logoKibana" size="xl" />,
+      title: <h2>Loading Dashboards</h2>,
```
```suggestion
      title: <h2>Loading dashboards</h2>,
```

miukimiu commented on src-docs/src/views/empty_prompt/guidelines.tsx:
```diff
@@ -27,174 +28,210 @@ export default () => (
+        title="Anatomy of an empty state"
+        width="1200"
+        height="550"
+        src="https://www.figma.com/embed?embed_host=share&url=https%3A%2F%2Fwww.figma.com%2Ffile%2FRzfYLj2xmH9K7gQtbSKygn%2FElastic-UI%3Fnode-id%3D22764%253A276515"
```
We decided to use the Figma iframe because it's easier to keep the images up to date. It's synced with the EUI Figma project.

cchaos commented on src-docs/src/views/empty_prompt/_types_of_use_cases.tsx:
```diff
@@ -0,0 +1,286 @@
+            For no data use cases, consider using a{' '}
+            <Link to="/display/card">EuiCard</Link>. In Kibana, you just need to
+            pass a no data configuration into your{' '}
+            <strong>KibanaPageTemplate</strong> to display a specific UI that
```
👍  Great idea! We could also link to this docs page in the Kibana repo: https://github.com/elastic/kibana/blob/main/dev_docs/tutorials/kibana_page_template.mdx

I thought about linking to the new docs site, but because it's still under dev and behind a login, I think we'll want a public link. Once the new docs site is live, we can point there instead.

gchaps replied:
LGTM

kibanamachine commented:
Preview documentation changes for this PR: https://eui.elastic.co/pr_5409/
